src/main/java/seedu/savenus/logic/commands/PreferenceCommand.java

```diff
@@ -0,0 +1,111 @@
+    private final Set<Location> locationList;
+
+    /**
+     * Creates an PreferenceCommand to add the user's recommendations
+     */
+    public PreferenceCommand(Set<Category> categoryList, Set<Tag> tagList, Set<Location> locationList) {
+        requireAllNonNull(categoryList, tagList, locationList);
+
+        // Convert all to lowercase
+        this.categoryList = categoryList.stream()
+                .map(c -> new Category(c.category.toLowerCase())).collect(Collectors.toSet());
+        this.tagList = tagList.stream()
+                .map(t -> new Tag(t.tagName.toLowerCase())).collect(Collectors.toSet());
+        this.locationList = locationList.stream()
+                .map(l -> new Location(l.location.toLowerCase())).collect(Collectors.toSet());
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        throw new AssertionError("This method should not be called.");
+    }
+
+    /**
+     * Executes the preference command.
+     *
+     * @param model  The current model
+     * @param isLike True if adding likes or false if adding dislikes
+     * @return A success message including the list of likes and dislikes
+     */
+    public CommandResult execute(Model model, boolean isLike) throws CommandException {
+        StringBuilder result = new StringBuilder();
+
+        RecommendationSystem recommendationSystem = model.getRecommendationSystem();
+
+        if (isLike) {
+            // Throws a command exception if any of the likes are in dislikes or vice versa
+            if (recommendationSystem.getDislikedCategories().stream().anyMatch(categoryList::contains)
+                    || recommendationSystem.getDislikedLocations().stream().anyMatch(locationList::contains)
+                    || recommendationSystem.getDislikedTags().stream().anyMatch(tagList::contains)) {
+                throw new CommandException(DUPLICATE_FOUND_IN_OPPOSITE_LIST);
+            }
+
+            model.addLikes(categoryList, tagList, locationList);
+            result.append(" Liked: ");
+        } else {
+            // Throws a command exception if any of the likes are in dislikes or vice versa
+            if (recommendationSystem.getLikedCategories().stream().anyMatch(categoryList::contains)
+                    || recommendationSystem.getLikedLocations().stream().anyMatch(locationList::contains)
+                    || recommendationSystem.getLikedTags().stream().anyMatch(tagList::contains)) {
+                throw new CommandException(DUPLICATE_FOUND_IN_OPPOSITE_LIST);
+            }
+
+            model.addDislikes(categoryList, tagList, locationList);
+            result.append(" Disliked: ");
+        }
+
+        String addedItems = "Categories: " + Arrays.toString(categoryList.toArray())
+                + " Tags: " + Arrays.toString(tagList.toArray())
+                + " Locations: " + Arrays.toString(locationList.toArray()) + "\n";
+
+        result.append(addedItems);
```

Comment:
Maybe its possible to just have 
_Success! Liked: Chinese
Current likes: Categories: [Chinese] Tags: [] Locations: []
Current dislikes: Categories: [] Tags: [] Locations: []_
instead of 
_Success! Liked: Categories: [Chinese] Tags: [] Locations: []
Current likes: Categories: [Chinese] Tags: [] Locations: []
Current dislikes: Categories: [] Tags: [] Locations: []_
as it might be slightly repetitive to the user.

Reply:
Done!